### PR TITLE
Redis test environment container

### PR DESCRIPTION
**Pull request by kumarnmanoj**
# Redis Test environment container

Added to test the systems that are using redis.

### CODE REVIEW

**Comment by kumarnmanoj** [comment]
@Deffiss Required your expedited support. We are already using the existing containers (mongo, kafka) in our integration tests. Redis is the only missing piece.

**Comment by Deffiss** [comment]
Hi, @kumarnmanoj , thanks for the contribution. I'll start the review.

File: src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                EndPoints =
+                {
+                    { redisConnectionConfiguration.Host, redisConnectionConfiguration.Port },
+                },
+                Password = redisConnectionConfiguration.Password
+            };
+
+            var redis = await ConnectionMultiplexer.ConnectAsync(redisConfigurationOptions);
+
+            var s = redis.GetStatus();
```
Comment (Deffiss) [inline src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs:38]: Do you need to check the status here or it is really meaningless and just successful response is enough?

File: src/TestEnvironment.Docker.Containers.Redis/TestEnvironment.Docker.Containers.Redis.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Deffiss) [inline src/TestEnvironment.Docker.Containers.Redis/TestEnvironment.Docker.Containers.Redis.csproj:1]: Looks like tabs style is not consistent in this file.

File: test/TestEnvironment.Docker.Tests/TestEnvironment.Docker.Tests.csproj [hunk 3]
```diff
@@ -35,7 +35,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\src\TestEnvironment.Docker.Containers.Kafka\TestEnvironment.Docker.Containers.Kafka.csproj" />
+	<ProjectReference Include="..\..\src\TestEnvironment.Docker.Containers.Redis\TestEnvironment.Docker.Containers.Redis.csproj" />
```
Comment (Deffiss) [inline test/TestEnvironment.Docker.Tests/TestEnvironment.Docker.Tests.csproj:38]: Again, tabs/spaces

**Review by Deffiss — CHANGES_REQUESTED** [review]

**Comment by Deffiss** [comment]
@kumarnmanoj - few minor comments

File: src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs (hunk 1 above)
Comment (kumarnmanoj) [reply src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs:38]: If there is connection/password issue, connectAsync will throw exception. Once the connectAsync passed it is safe to assume that the redis is up and running. In addition to that GetStatus will also give successful response if connection is successful.

This option we have is safe.

Comment (Deffiss) [reply src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs:38]: In that case you can replace `s` variable with `_`.

Comment (kumarnmanoj) [reply src/TestEnvironment.Docker.Containers.Redis/RedisContainerWaiter.cs]: Introduced the discard variable.

File: test/TestEnvironment.Docker.Tests/TestEnvironment.Docker.Tests.csproj (hunk 3 above)
Comment (kumarnmanoj) [reply test/TestEnvironment.Docker.Tests/TestEnvironment.Docker.Tests.csproj]: Fixed the styles

**Review by Deffiss — APPROVED** [review]

**Comment by kumarnmanoj** [comment]
@Deffiss Thank you for the review!. The PR is good for merging?

**Comment by Deffiss** [comment]
Yes, merging it.

**Comment by Deffiss** [comment]
https://www.nuget.org/packages/TestEnvironment.Docker.Containers.Redis - will be published soon

**Pull request merged**